src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -202,19 +202,18 @@ internal static TarEntryType GetCorrectTypeFlagForFormat(TarEntryFormat format,
         /// <summary>Parses a byte span that represents an ASCII string containing a number in octal base.</summary>
         internal static T ParseOctal<T>(ReadOnlySpan<byte> buffer) where T : struct, INumber<T>
         {
-            buffer = TrimEndingNullsAndSpaces(buffer);
-
             T octalFactor = T.CreateTruncating(8u);
             T value = T.Zero;
-            foreach (byte b in buffer)
+
+            // skip leading non-octal bytes
+            int offset = 0;
+            for (; offset < buffer.Length && (buffer[offset] < (byte)'0' || buffer[offset] > (byte)'7'); ++offset);
+
+            foreach (byte b in buffer.Slice(offset))
             {
-                uint digit = (uint)(b - '0');
-                if (digit >= 8)
-                {
-                    ThrowInvalidNumber();
-                }
+                if (b < (byte)'0' || b > (byte)'7') break;
 
-                value = checked((value * octalFactor) + T.CreateTruncating(digit));
+                value = checked((value * octalFactor) + T.CreateTruncating((uint)(b - '0')));
```

Comment:
Which assets have octal fields with non-octal characters on either end? I'm concerned that we are basically allowing garbage and not respecting the tar spec anymore.

Reply:
Yea let's not anywhere intentionally be more lenient than we discover existing tools are.

Reply:
@danmoseley, it is exactly as lenient as libarchive's octal parsing, it allows any non-octal character at the beginning and end.

GNU tar and busybox implementations allow both leading and trailing whitespaces and nulls.

.NET's implementation is the most strict one, we are only ignoring null and spaces when they appear at the end, and we do not allow them at the beginning.

I haven't find specification calling out for and against these behaviors, so if the formal specifications do not call it out explicitly, then it's up to us to decide which existing implementation is to follow.

Reply:
But ignoring nulls and spaces is not the same as ignoring _any_ character that is either smaller than '0' or larger than '7'.

Reply:
libarchive ignores _any_ non-octal character. :)
https://github.com/libarchive/libarchive/blob/9e5279615033fa073d0bbae83b366359eab3b62f/contrib/untar.c#L45

if this is too liniment, I can confine it to only allow nulls and spaces.